src/generators/npm-install.ts

```diff
@@ -0,0 +1,30 @@
+import { readPackageJSON } from "pkg-types";
+import { codeBlock } from "omark";
+import { defineGenerator } from "../generator";
+
+export default defineGenerator({
+  name: "npm-install",
+  async generate({ options, args }) {
+    const pkg = await readPackageJSON(options.dir);
+    const name = args.name || pkg.name || "package-name";
```

Comment:
- We only need to read packageJSON if no explicit `name` argument is provided. 
- For `readPackageJSON` it can be chained with `.catch(() => undefined)` to prevent making an error when there is no package
- By default we can early return to self-disable when no package name is found